### PR TITLE
feat: support envFrom reference in all components

### PR DESCRIPTION
**Pull request by drodriguez-305**
### Which problem does the PR fix?
Related to:

Task: https://github.com/camunda/camunda-platform-helm/issues/1677

### What's in this PR?
This PR adds support for including environment variables from ConfigMaps and Secrets using envFrom in the deployment configurations of various components. This change enhances the flexibility and security of the deployment process by allowing the injection of environment variables from external resources.

### Checklist

Please make sure to follow our [Contributing Guide](../blob/main/docs/contributing.md).

**Before opening the PR:**

- [x] In the repo's root dir, run `make go.update-golden-only`.
- [x] There is no other open [pull request](../pulls) for the same update/change.
- [ ] Tests for charts are added (if needed).
- [ ] In-repo [documentation](../blob/main/docs/contributing.md#documentation) are updated (if needed).

**After opening the PR:**

- [x] Did you sign our CLA (Contributor License Agreement)? It will show once you open the PR.
- [ ] Did all checks/tests pass in the PR?

### CODE REVIEW

File: charts/camunda-platform/values.yaml [hunk 1]
```diff
@@ -555,6 +557,7 @@ console:
   ## @param console.affinity can be used to define pod affinity or anti-affinity https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity
   affinity: {}
 
+
```
Comment (aabouzaid) [inline charts/camunda-platform/values.yaml:560]: Extra.

File: charts/camunda-platform/templates/connectors/deployment.yaml [hunk 2]
```diff
@@ -85,11 +85,14 @@ spec:
           {{- if .Values.connectors.env}}
             {{ .Values.connectors.env | toYaml | nindent 12 }}
           {{- end }}
-          {{- if .Values.global.identity.auth.enabled }}
           envFrom:
+          {{- if .Values.global.identity.auth.enabled }}
             - configMapRef:
                 name: {{ include "camundaPlatform.fullname" . }}-identity-env-vars
           {{- end }}
+          {{- if .Values.connectors.envFrom }}
```
Comment (aabouzaid) [inline charts/camunda-platform/templates/connectors/deployment.yaml:93]: It should be `{{- if .Values.connectors.envFrom -}}` (with minus at the end) to remove the extra line before the if.

File: charts/camunda-platform/test/unit/console/golden/deployment.golden.yaml [hunk 3]
```diff
@@ -69,6 +69,7 @@ spec:
               value: console
             - name: CAMUNDA_CONSOLE_CONTEXT_PATH
               value: ""
+          envFrom:
```
Comment (aabouzaid) [inline charts/camunda-platform/test/unit/console/golden/deployment.golden.yaml:72]: Let's have an if condition for the components that don't have preconfigure `envFrom` so we keep the golden/rendered files clean.

**Review by aabouzaid — CHANGES_REQUESTED** [review]

File: charts/camunda-platform/values.yaml (hunk 1 above)
Comment (drodriguez-305) [reply charts/camunda-platform/values.yaml]: fixed

File: charts/camunda-platform/test/unit/console/golden/deployment.golden.yaml (hunk 3 above)
Comment (drodriguez-305) [reply charts/camunda-platform/test/unit/console/golden/deployment.golden.yaml]: Moved the if lines above `envFrom` the golden files are set now.

File: charts/camunda-platform/templates/connectors/deployment.yaml (hunk 2 above)
Comment (drodriguez-305) [reply charts/camunda-platform/templates/connectors/deployment.yaml]: The ones that needed this were the ones that had the preconfigured `envFrom` the others I did not add.

**Review by aabouzaid — APPROVED** [review]
Great job @drodriguez-305 :raised_hands:

**Pull request merged**